src/main/java/com/minecolonies/coremod/network/messages/ReplaceBlockMessage.java

```diff
@@ -0,0 +1,211 @@
+                        final ItemStack stackToPlace = message.blockTo.copy();
+                        stackToPlace.setCount(stackToPlace.getMaxStackSize());
+                        fakePlayer.setHeldItem(EnumHand.MAIN_HAND, stackToPlace);
+
+                        if (message.blockTo.getItem() instanceof ItemBed)
+                        {
+                            fakePlayer.rotationYaw = blockState.getValue(BlockBed.FACING).getHorizontalIndex() * 90;
+                        }
+                        final EnumFacing facing = (message.blockTo.getItem() instanceof ItemDoor
+                                || message.blockTo.getItem() instanceof ItemBed
+                                || message.blockTo.getItem() instanceof ItemSlab)? EnumFacing.UP : EnumFacing.NORTH;
+                        ForgeHooks.onPlaceItemIntoWorld(stackToPlace, fakePlayer, world, here, facing, 0, 0, 0, EnumHand.MAIN_HAND);
+
+                        final IBlockState newBlockState= world.getBlockState(here);
+                        if (newBlockState.getBlock() instanceof BlockStairs && blockState.getBlock() instanceof BlockStairs)
+                        {
+                            IBlockState transformation = newBlockState.withProperty(BlockStairs.FACING, blockState.getValue(BlockStairs.FACING));
+                            transformation = transformation.withProperty(BlockStairs.HALF, blockState.getValue(BlockStairs.HALF));
+                            transformation = transformation.withProperty(BlockStairs.SHAPE, blockState.getValue(BlockStairs.SHAPE));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockHorizontal && blockState.getBlock() instanceof BlockHorizontal
+                                && !(blockState.getBlock() instanceof BlockBed))
+                        {
+                            final IBlockState transformation = newBlockState.withProperty(BlockHorizontal.FACING, blockState.getValue(BlockHorizontal.FACING));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockDirectional && blockState.getBlock() instanceof BlockDirectional)
+                        {
+                            final IBlockState transformation = newBlockState.withProperty(BlockDirectional.FACING, blockState.getValue(BlockDirectional.FACING));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockSlab && blockState.getBlock() instanceof BlockSlab)
+                        {
+                            final IBlockState transformation;
+                            if (blockState.getBlock() instanceof BlockDoubleStoneSlab || blockState.getBlock() instanceof BlockDoubleStoneSlabNew)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)